### PR TITLE
fix(futures) Allow calling ThreadPoolExecutor.submit with fn kwarg

### LINKED ISSUE #3000: tuple index out of range of threading.py
After upgrading from ddtrace==0.46.0 to version ddtrace==0.55.4 my service crash with IndexError.
```
Traceback (most recent call last):
  File "/my_service/services/base_service.py", line 105, in run
    futures.append(executor.submit(fn=self._single_entry_point_run, entry_point=entry_point))
  File "/my_service/venv/lib/python3.7/site-packages/ddtrace/contrib/futures/threading.py", line 26, in _wrap_submit
    fn = args[0]
IndexError: tuple index out of range
```

I'm facing this issue even when setting futures=False.
`patch_all(celery=True, django=True, psycopg2=True, redis=True, futures=True)`

### PR DESCRIPTION
**Pull request by brettlangdon**
## Commit Message
{{title}}

Calling `ThreadPoolExecutor.submit(fn=target)` would raise an `IndexError: tuple index out of range` exception.

This is because we assumed that `args[0]` would always be the target function and never considered that it could be provided in `kwargs`.

We are not using our helper function because we need to remove `fn` from either `args` or `kwargs` once it is found.

Fixes #3000

### CODE REVIEW

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.56

File: ddtrace/contrib/futures/threading.py [hunk 1]
```diff
@@ -21,10 +21,12 @@ def _wrap_submit(func, instance, args, kwargs):
     if ddtrace.tracer.context_provider._has_active_context():
         current_ctx = ddtrace.tracer.context_provider.active()
 
-    # extract the target function that must be executed in
-    # a new thread and the `target` arguments
-    fn = args[0]
-    fn_args = args[1:]
+    # The target function can be provided as a kwarg argument "fn" or the first positional argument
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/futures/threading.py:24]: I was going to suggest using `get_argument_value` but we do unconventional things with the positional args so this is :+1: by me

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Comment by majorgreys** [comment]
Fixes #3000

**Comment by majorgreys** [comment]
@Mergifyio backport 0.56

**Pull request merged**